homeassistant/components/tplink/utils.py

```diff
@@ -0,0 +1,13 @@
+"""Common code for tplink."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+
+@callback
+def async_entry_is_legacy(entry: ConfigEntry) -> bool:
```

Comment:
Should we keep this local to its callsite? Considering that this file will be removed after the migration code gets removed in a couple of releases, I'm unsure if it's worth having it in a separate file until then.

Reply:
Since it is used in both the config flow and init, it seemed like it should live outside of both.  If you want to move it to init, I think thats ok as well.  Its kinda *meh* since we are going to delete it soon anyways.

Reply:
71271dcee8